core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -365,55 +365,65 @@ class WorkspaceService(protected val userInfo: UserInfo,
     }
   }
 
-  private def deleteWorkspace(workspaceName: WorkspaceName, workspaceContext: Workspace): Future[PerRequestMessage] = {
-    //Attempt to abort any running workflows so they don't write any more to the bucket.
-    //Notice that we're kicking off Futures to do the aborts concurrently, but we never collect their results!
-    //This is because there's nothing we can do if Cromwell fails, so we might as well move on and let the
-    //ExecutionContext run the futures whenever
-    val deletionFuture: Future[Seq[WorkflowRecord]] =
-      requesterPaysSetupService.revokeAllUsersFromWorkspace(workspaceContext).flatMap { _ =>
-        dataSource.inTransaction { dataAccess =>
-          for {
-            // Gather any active workflows with external ids
-            workflowsToAbort <- dataAccess.workflowQuery.findActiveWorkflowsWithExternalIds(workspaceContext)
-
-            //If a workflow is not done, automatically change its status to Aborted
-            _ <- dataAccess.workflowQuery.findWorkflowsByWorkspace(workspaceContext).result.map { recs =>
-              recs.collect {
-                case wf if !WorkflowStatuses.withName(wf.status).isDone =>
-                  dataAccess.workflowQuery.updateStatus(wf, WorkflowStatuses.Aborted) { status =>
-                    if (config.trackDetailedSubmissionMetrics) Option(workflowStatusCounter(workspaceSubmissionMetricBuilder(workspaceName, wf.submissionId))(status))
-                    else None
-                  }
+  private def gatherWorkflowsToAbortAndSetStatusToAborted(workspaceName: WorkspaceName, workspaceContext: Workspace) = {
+    dataSource.inTransaction { dataAccess =>
+      for {
+        // Gather any active workflows with external ids
+        workflowsToAbort <- dataAccess.workflowQuery.findActiveWorkflowsWithExternalIds(workspaceContext)
+
+        //If a workflow is not done, automatically change its status to Aborted
+        _ <- dataAccess.workflowQuery.findWorkflowsByWorkspace(workspaceContext).result.map { recs =>
+          recs.collect {
+            case wf if !WorkflowStatuses.withName(wf.status).isDone =>
+              dataAccess.workflowQuery.updateStatus(wf, WorkflowStatuses.Aborted) { status =>
+                if (config.trackDetailedSubmissionMetrics) Option(workflowStatusCounter(workspaceSubmissionMetricBuilder(workspaceName, wf.submissionId))(status))
```

Comment:
You're not doing anything with the result of `collect` - are these functions (that all read like nouns) intended for side effects??

Reply:
looks like they are used for metrics